data_processing/market/eia.py

```diff
@@ -0,0 +1,2087 @@
+
+                    global_dict.update({name_old: name_dict})
+
+                    if (name_old, region_old) not in names_checked:
+
+                        names_checked.append((name_old, region_old))
+
+                elif name_old in global_dict:
+
+                    if (name_old, region_old) in names_checked:
+
+                        counter = 0
+
+                        for fuel in name_dict:
+
+                            for prime in name_dict[fuel]:
+
+                                counter += name_dict[fuel][prime]
+
+                                if fuel in global_dict[name_old]:
+
+                                    if prime in global_dict[name_old][fuel]:
+
+                                        global_dict[name_old][fuel][prime] += (
+                                            name_dict[fuel][prime]
+                                        )
+
+                                    else:
+
+                                        global_dict[name_old][fuel].update(
+                                            {prime: name_dict[fuel][prime]}
+                                        )
+
+                                elif fuel not in global_dict[name_old]:
+
+                                    global_dict[name_old].update(
+                                        {fuel: name_dict[fuel]}
+                                    )
+
+                                new_total = (
+                                    counter + (
+                                        global_dict[name_old][
+                                            'Total Generation'
+                                        ]
+                                    )
+                                )
+
+                                global_dict[name_old].update(
+                                    {'Total Generation': new_total}
+                                )
+
+                    elif (name_old, region_old) not in names_checked:
+
+                        name_old = name_old + ' ' + region_old
+
+                        counter = 0
+
+                        for fuel in name_dict:
+
+                            for prime in name_dict[fuel]:
+
+                                counter += name_dict[fuel][prime]
+
+                        name_dict.update({'Total Generation': counter})
+
+                        global_dict.update({name_old: name_dict})
+
+                name_dict = dict()
+
+                if fuel_type in name_dict:
+
+                    if mover in name_dict[fuel_type]:
+
+                        name_dict[fuel_type][mover] += amount
+
+                    elif mover not in name_dict[fuel_type]:
+
+                        name_dict[fuel_type].update({mover: amount})
+
+                elif fuel_type not in name_dict:
+
+                    name_dict.update({fuel_type: {mover: amount}})
+
+            name_old = name
+
+            region_old = region
+
+    del global_dict['']
```

Comment:
@katyhuff  posted in the pro-tips slack channel about leaving "generous amounts of comments in your code." 

This is a great example as to why. 

Also, the way I understand it, you're manipulating a ``.csv`` file and turning it into a "nested dictionary." I think this can and should be done using the ``pandas`` library. If it's a higher dimensional dataframe, then I suggest using ``XArray``. 

The benefit of using these powerful libraries is that it makes your code much more comprehensible. If we recall the Zen of Python: Namespaces are a honking great idea.